common/app/routes/challenges/components/Show.jsx

```diff
@@ -87,13 +95,16 @@ export class Challenges extends PureComponent {
   }
 
   componentWillReceiveProps(nextProps) {
+    const { title } = nextProps;
     const { block, dashedName } = nextProps.params;
     const { resetUi, updateTitle, replaceChallenge } = this.props;
     if (this.props.params.dashedName !== dashedName) {
-      updateTitle(nextProps.title);
       resetUi();
       replaceChallenge({ dashedName, block });
     }
+    if (this.props.title !== title) {
```

Comment:
Sounds like this has more to do with the title not being set correctly in the first place.